Rubberduck.Parsing/Symbols/Declaration.cs

```diff
@@ -159,25 +159,31 @@ public class Declaration : IEquatable<Declaration>
 
             _projectId = _qualifiedName.QualifiedModuleName.ProjectId;
 
-            var @namespace = Annotations.FirstOrDefault(annotation => annotation.AnnotationType == AnnotationType.Folder);
-            string result;
-            if (@namespace == null)
-            {
-                result = _qualifiedName.QualifiedModuleName.Project == null
-                    ? _projectId
-                    : _qualifiedName.QualifiedModuleName.Project.Name;
-            }
-            else
-            {
-                var value = ((FolderAnnotation)@namespace).FolderName;
-                result = value;
-            }
-            _customFolder = result;
+            _customFolder = FolderFromAnnotations();
             _isArray = isArray;
             _asTypeContext = asTypeContext;
             _typeHint = typeHint;
         }
 
+            private string FolderFromAnnotations()
+            {
+                var @namespace = Annotations.FirstOrDefault(annotation => annotation.AnnotationType == AnnotationType.Folder);
+                string result;
+                if (@namespace == null)
+                {
+                    result = _qualifiedName.QualifiedModuleName.Project == null
```

Comment:
this could be cleaner with null-unwrapping.. do we even have that language level?? @retailcoder